### PR TITLE
Remove double encoding for config and set content-type

### LINKED ISSUE #94: Use JSON response content type for GET /config
Currently the `GET /config` call to the management API returns a serialized JSON with `text/plain` content type. It would be better to return it as `application/json`.

### PR DESCRIPTION
**Pull request by mayorova**
Closes #94 

- `provider.contents` is already JSON-encoded, so no need to encode it again.
- adding `application/json` content type instead of the default `text/plain`

### CODE REVIEW

**Comment by mikz** [comment]
There is a test case covering it. https://github.com/3scale/apicast/blob/f0f7e8e218dcf519d40135c48217660481a425b2/t/001-management.t#L115-L132
Looks like the contents could be different based on from where it is loaded?
Also some tests fail because of `nil` in lua vs `null` in json. It should do the conversion at least when it is nil so it returns correct `null`.

I guess it should convert it to json when it is not string. Maybe we need some other test case for this.

File: t/001-management.t [hunk 1]
```diff
@@ -244,3 +246,26 @@ null
...
+
+=== TEST 9: all endpoints use correct Content-Type
+JSON response body and content type application/json should be returned.
+--- http_config
+  lua_package_path "$TEST_NGINX_LUA_PATH";
+--- config
+  include $TEST_NGINX_MANAGEMENT_CONFIG;
+--- request eval
+[ "DELETE /config", "PUT /config 
+{\"services\":[{\"id\":42}]}", "POST /config
```
Comment (mikz) [inline t/001-management.t]: might be cool to use `'` here, to get rid of the escaping

**Comment by mikz** [comment]
Awesome!

**Comment by hallelujah** [comment]
:+1: Looks really good!
